### PR TITLE
Add LICENSE to tars

### PR DESCRIPTION
**Pull request by Jaanki**
Add LICENSE file to the subctl binary tars. This is a requirement to enable krew installation of subctl.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -94,11 +94,11 @@ cmd/bin/subctl: cmd/bin/subctl-$(VERSION)-$(GOOS)-$(GOARCH)$(GOEXE)
 
 dist/subctl-%.tar.xz: cmd/bin/subctl-%
 	mkdir -p dist
-	tar -cJf $@ --transform "s|^cmd/bin/subctl-[^/]*|subctl-$(VERSION)/subctl|" $<
+	tar -cJf $@ --transform "s|^cmd/bin/subctl-[^/]*|subctl-$(VERSION)/subctl|" --add-file=LICENSE $<
```
Comment (skitt) [inline Makefile]: The `LICENSE` file ends up being a prerequisite for the archive, list it as such and change the recipe to include all prerequisites:

```make
dist/subctl-%.tar.xz: cmd/bin/subctl-% LICENSE
	mkdir -p dist
	tar -cJf $@ --transform "s|^cmd/bin/subctl-[^/]*|subctl-$(VERSION)/subctl|" $^
```

File: Makefile [hunk 2]
```diff
@@ -94,11 +94,11 @@ cmd/bin/subctl: cmd/bin/subctl-$(VERSION)-$(GOOS)-$(GOARCH)$(GOEXE)
...
 
 dist/subctl-%.tar.gz: cmd/bin/subctl-%
 	mkdir -p dist
-	tar -czf $@ --transform "s|^cmd/bin/subctl-[^/]*|subctl-$(VERSION)/subctl|" $<
+	tar -czf $@ --transform "s|^cmd/bin/subctl-[^/]*|subctl-$(VERSION)/subctl|" --add-file=LICENSE $<
```
Comment (skitt) [inline Makefile]: See above.

**Review by skitt — CHANGES_REQUESTED** [review]
Is the requirement for a `LICENSE` file described anywhere? Please add a link to relevant documentation if you can find it.

**Comment by Jaanki** [comment]
> Is the requirement for a `LICENSE` file described anywhere? Please add a link to relevant documentation if you can find it.

It is not mentioned anywhere but the CI job on [my PR in krew-index repository](https://github.com/kubernetes-sigs/krew-index/pull/3432) (which adds manifest file for subctl) is failing because of missing `LICENSE` file in the tar.

**Review by skitt — APPROVED** [review]

**Review by sridhargaddam — APPROVED** [review]

**Pull request merged**